iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "mlir/Dialect/Math/Transforms/Approximation.h"
+#include "mlir/Dialect/Math/Transforms/Passes.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+/// Command line to use native hardware operations instead of polynomial
+/// approximation.
+static llvm::cl::opt<bool> clNativeMathPrecision(
```

Comment:
Ill let you decide if you want to do this now or not, but just in terms of "longevity", this is very short lived. We should have all such command line options on the CUDA target backend (like you have for sm version i believe). Propogate that to the dictionary attribute on `hal.device.target` attribute and query it from there. This kind of flags for long lived features (and I suspect this will be) is discouraged

Reply:
I'm not super clear on what should be the long term options for this so I would rather wait before adding extra infrastructure. I think we can keep this as just a way to quickly evaluate perf impact of this lowering and we can remove it or make it something non experimental later. What do you think? Do you think it needs more comments to clarify this part?

Reply:
sure. sounds good. As long as you are signed up to either deprecate this or move it out of here to something long term.

Reply:
sounds good :)